pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -1069,13 +993,82 @@ func (npc *NetworkPolicyController) syncPodFirewallChains(networkPoliciesInfo []
 				return nil, fmt.Errorf("Failed to run iptables command: %s", err.Error())
 			}
 		}
+	}
+
+	// loop through the pods running on the node which to which ingress network policies to be applied
+	ingressNetworkPolicyEnabledPods, err := npc.getIngressNetworkPolicyEnabledPods(networkPoliciesInfo, npc.nodeIP.String())
```

Comment:
My understanding is that there is some benefit to having `Established,Related` traffic which accounts for the bulk of traffic and is an easy determination for iptables at the top of the chain. If I were you, I would move the Ingress and Egress chain handling near the top of this function so that when you do the calls to `Insert` the `--ctstate RELATED,ESTABLISHED` at position 1, it will come in above these rules and get processed in the chain first and ACCEPTED before iptables has to made determinations about these chains.